Sources/Purchasing/Purchases/Purchases.swift

```diff
@@ -1743,21 +1743,23 @@ private extension Purchases {
         // Note: it's important that we observe "will enter foreground" instead of
         // "did become active" so that we don't trigger cache updates in the middle
         // of purchases due to pop-ups stealing focus from the app.
-        self.updateAllCachesIfNeeded(isAppBackgrounded: false)
-        self.dispatchSyncSubscriberAttributes()
+        self.operationDispatcher.dispatchOnWorkerThread {
```

Comment:
I really dislike how indirect this is, but it's not a bad idea to dispatch this work on the worker thread in any case

Reply:
I'm slightly concerned about any unknown side-effects... but nothing comes to mind right now. I think this should be safe yeah.

Reply:
I believe it should be okay, all this code is already meant to be working on its own without user interaction (other than the user opening the SDK, that is)